### PR TITLE
Add compose param expression tool

### PR DESCRIPTION
**Pull request by mvdbeek**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mvdbeek** [comment]
This tool should be universally useful and I was considering adding it to galaxy proper, but I think we'll want to be able to have multiple versions of installed, should we change the tool in the future.

**Comment by mvdbeek** [comment]
Awesome, green on first try!

**Comment by mvdbeek** [comment]
In case you're wondering what sort of things can be done with this, here's a workflow that takes to output of `samtools view -c` and uses that as the nominator in the `Compute an expression on every row` tool:

Screen Shot 2019-05-13 at 09 03 00

**Comment by mblue9** [comment]
I was wondering and was going to ask 😄. Do you think you could put a bit more info in the help for users so they'd know when to use it and/or make a mini-tutorial that demonstrates what it does at some point?

**Comment by mvdbeek** [comment]
Definitely planning on a tutorial, if you have suggestions for the help text let me know, it's a bit hard to express how this can be used without an example.

File: tools/compose_text_param/compose_text_param.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            </repeat>
+            <output name="output">
+                <assert_contents> 
+                    <!-- <has_line line="'"Text: value, Integer: 1, Float: 1.5"'"/> -->
+                    <has_line line="&quot;Text: value, Integer: 1, Float: 1.5&quot;"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help>
```
Comment (mblue9) [inline tools/compose_text_param/compose_text_param.xml]: Should this help have CDATA tags?

**Comment by mblue9** [comment]
Yeah I think seeing an example would be best so maybe a link to the tutorial (or an example workflow) could be added to the help when you have it.

File: tools/compose_text_param/compose_text_param.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    <param name="component_value" value="" type="integer" label="Enter integer that should be part of the computed value"/>
+                </when>
+                 <when value="float">
+                    <param name="component_value" value="" type="float" label="Enter float that should be part of the computed value"/>
+                </when>
+            </conditional>
+        </repeat>
+    </inputs>
+    <outputs>
+        <output type="text" name="out1" from="output" />
```
[the remark this answers is not in the document]
Comment (mvdbeek) [reply tools/compose_text_param/compose_text_param.xml]: Yes, it's always needed in Galaxys execution model. Among other things this is how workflows are connected.

**Pull request merged**